diff --git a/bin/unpack-sources b/bin/unpack-sources
index 7221696e..9d91e06e 100755
--- a/bin/unpack-sources
+++ b/bin/unpack-sources
@@ -87,5 +87,5 @@ for tarball in $tarballs ; do
     # create symlinks for module directories; ignore git-hooks directory
     while read -r dir; do
         ln -sf "src/${tarname}/$(basename "$dir")" "$start_dir"
-    done < <(find "$lo_src_dir/$tarname" -mindepth 1 -maxdepth 1 -type d -path $lo_src_dir/$tarname/git-hooks)
+    done < <(find "$lo_src_dir/$tarname" -mindepth 1 -maxdepth 1 -type d ! -path $lo_src_dir/$tarname/git-hooks)
 done
