diff --git a/sd/source/ui/unoidl/unoobj.cxx b/sd/source/ui/unoidl/unoobj.cxx
index fc6864250..353818bc4 100644
--- a/sd/source/ui/unoidl/unoobj.cxx
+++ b/sd/source/ui/unoidl/unoobj.cxx
@@ -773,9 +773,24 @@ void SAL_CALL SdXShape::setPropertyValue( const OUString& aPropertyName, const c
             u"LineDashName"_ustr,
             u"LineStartName"_ustr,
             u"LineEndName"_ustr,
+            u"FillColor"_ustr,
+            u"FillGradientName"_ustr,
+            u"FillHatchName"_ustr,
+            u"FillStyle"_ustr,
             u"FillTransparence"_ustr,
             u"LineTransparence"_ustr,
             u"RotateAngle"_ustr,
+            u"GlowEffectRadius"_ustr,
+            u"GlowEffectColor"_ustr,
+            u"GlowEffectTransparency"_ustr,
+            u"SoftEdgeRadius"_ustr,
+            u"EdgeKind"_ustr,
+            u"StartPosition"_ustr,
+            u"EndPosition"_ustr,
+            u"EdgeStartConnection"_ustr,
+            u"EdgeEndConnection"_ustr,
+            u"EdgeStartGluePointIndex"_ustr,
+            u"EdgeEndGluePointIndex"_ustr,
         };
         SdrObject* pObject = mpShape->GetSdrObject();
         SdDrawDocument* pDocument = mpModel ? mpModel->GetDoc() : nullptr;

