diff --git a/sd/source/ui/view/drviews7.cxx b/sd/source/ui/view/drviews7.cxx
index f4373e602bda5bafc335dd21fbf35a0956ee931b..7f6a3510bd53c66bd42fef40bcd68440abd2632e 100644
--- a/sd/source/ui/view/drviews7.cxx
+++ b/sd/source/ui/view/drviews7.cxx
@@ -87,6 +87,7 @@
 #include <fuformatpaintbrush.hxx>
 #include <stlsheet.hxx>
 #include <SlideSectionManager.hxx>
+#include <undoback.hxx>
 
 using namespace ::com::sun::star;
 using namespace ::com::sun::star::uno;
@@ -1758,6 +1759,11 @@ void DrawViewShell::SetPageProperties (SfxRequest& rReq)
         SfxItemSet aTempSet = aPageItemSet.CloneAsValue(false, &mpDrawView->GetModel().GetItemPool());
         const SfxPoolItem* pItem = nullptr;
 
+        if (GetDoc()->IsUndoEnabled())
+            GetDocSh()->GetUndoManager()->AddUndoAction(
+                std::make_unique<SdBackgroundObjUndoAction>(
+                    *GetDoc(), *pPage, aPageItemSet), true);
+
         rPageProperties.ClearItem(XATTR_FILLSTYLE);
         rPageProperties.ClearItem(XATTR_FILLGRADIENT);
         rPageProperties.ClearItem(XATTR_FILLHATCH);
